js/events/visibility_change.js

```diff
@@ -0,0 +1,22 @@
+import $ from '../core/renderer';
+import { triggerHandler } from './core/events_engine';
+
+const triggerVisibilityChangeEvent = function(eventName) {
+    const VISIBILITY_CHANGE_SELECTOR = '.dx-visibility-change-handler';
+
+    return function(element) {
+        const $element = $(element || 'body');
+
+        const changeHandlers = $element.filter(VISIBILITY_CHANGE_SELECTOR).
+            add($element.find(VISIBILITY_CHANGE_SELECTOR));
+
+        for(let i = 0; i < changeHandlers.length; i++) {
+            triggerHandler(changeHandlers[i], eventName);
+        }
+    };
+};
+
+
+exports.triggerShownEvent = triggerVisibilityChangeEvent('dxshown'); // TODO: extract to events
```

Comment:
I'm not sure that increasing of TODOs is good idea. We already have [170 TODOs](https://github.com/DevExpress/DevExtreme/search?q=TODO) When it will be fixed? I suggest to use internal planning tool to list improvements

Reply:
I totally agree about that. But I was going to get rid of them in next PR. I just decided that this one is too large to do it.

Reply:
I was talking about [these TODOs](https://github.com/DevExpress/DevExtreme/pull/13021/files#diff-1ff023ae34a7f57e07f146626ad4e5d0R142). But I've just noticed that this comment is about another ones, which I forgot to remove. [Fixed](https://github.com/DevExpress/DevExtreme/pull/13021/commits/d2dbf2d53f76bb0ef3d0f1d75d0dabe43ca2bc7f).